### PR TITLE
Jscodeshift i18n

### PR DESCRIPTION
**Pull request by roshkins**
Added autoi18n util

usage: "npm run autoi18n -- /path/to/module.js"

space after '--' important

Closes #3896

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     "coverage": "cross-env NODE_ENV=test nyc ava",
     "report-coverage": "nyc report --reporter=text-lcov | coveralls",
     "test-integration": "nightwatch --env",
-    "version": "node build/version.js"
+    "version": "node build/version.js",
+    "autoi18n": "jscodeshift --parser flow --transform utils/i18nTransformer.js "
```
Comment (BenMcGarry) [inline package.json]: Space between .js and ".

File: package.json [hunk 2]
```diff
@@ -111,6 +112,7 @@
     "inert-entry-webpack-plugin": "1.1.4",
     "interpolate-loader": "0.0.6",
     "jpm": "1.2.2",
+    "jscodeshift": "^0.3.30",
```
Comment (BenMcGarry) [inline package.json]: Please use specific versions.

File: utils/i18nTransformer.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                .replaceWith(descPath => {
+                    const descKey = `${modName}${upCase(optName)}Desc`; // use _.camelCase
+                    newI18nKeys[descKey] = descPath.node.value.type === 'TemplateLiteral' ?
+                        descPath.node.value.quasis[0].value.raw.trim() :
+                        descPath.node.value.value;
+                    return j.property('init', j.identifier('description'), j.literal(descKey));
+                });
+    	});
+
+			var fs = require('fs');
```
Comment (erikdesjardins) [inline utils/i18nTransformer.js]: All of these `var` can be `const` (and move this one to the top as mentioned)

File: utils/i18nTransformer.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+			var path = require("path");
+			const FileLocation = path.posix.resolve('./locales/locales/en.json');
+			var enJSON = fs.readFileSync( FileLocation, {'encoding': 'utf8'});
```
Comment (erikdesjardins) [inline utils/i18nTransformer.js]: Nit: strict camelcase like `enJson`, even for initialisms

File: utils/i18nTransformer.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+			var enJSONObj = JSON.parse(enJSON);
+
+			for (var key in newI18nKeys) {
```
Comment (erikdesjardins) [inline utils/i18nTransformer.js]: This loop can be replaced with `Object.assign(enJsonObj, newI18nKeys)`

File: package.json (hunk 2 above)
Comment (roshkins) [inline package.json]: thanks!

File: utils/i18nTransformer.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+				enJSONObj[key] = newI18nKeys[key];
+			}
+
+			var newENJSON = JSON.stringify(enJSONObj, null, '\t');
```
Comment (erikdesjardins) [inline utils/i18nTransformer.js]: You could inline a few of these variables that are only used once, but I don't particularly care here

File: utils/i18nTransformer.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                    newI18nKeys[descKey] = descPath.node.value.type === 'TemplateLiteral' ?
+                        descPath.node.value.quasis[0].value.raw.trim() :
+                        descPath.node.value.value;
+                    return j.property('init', j.identifier('description'), j.literal(descKey));
+                });
+    	});
+
+			var fs = require('fs');
+			var path = require("path");
+			const FileLocation = path.posix.resolve('./locales/locales/en.json');
```
Comment (roshkins) [inline utils/i18nTransformer.js]: Thanks for the canonicalization !

File: utils/i18nTransformer.js (hunk 4 above)
Comment (roshkins) [inline utils/i18nTransformer.js]: ok!

File: utils/i18nTransformer.js (hunk 6 above)
Comment (roshkins) [inline utils/i18nTransformer.js]: Yes I can. I did that to make debugging easier.

**Pull request merged**